### ISSUE TITLE
Preferred Provider: 1G is not in the permissible list of providers for this user and operation.

### ISSUE THREAD

**Issue by joshdhenry** [Travelport-Ukraine/uapi-json#313]
I am trying the below request but I get the error "Preferred Provider: 1G is not in the permissible list of providers for this user and operation."  I think my preferred provider needs to be 1P. Is there any way to change the preferred provider?

```
const settings = 
    { 
        auth: {
            username: 'my username', 
            password: 'my password', 
            targetBranch: 'my target branch',
        },
        production: false,
        debug: 2,
    };

const AirService = uAPI.createAirService(settings);

const params = {
  legs: [
    {
      from: 'LWO',
      to: 'JKT',
      departureDate: '2018-07-18'
    },
    {
      from: 'JKT',
      to: 'IEV',
      departureDate: '2018-07-21',
    },
  ],
  passengers: {
    ADT: 1,
  },
  cabins: ['Economy'], 
  requestId: '4e2fd1f8-2221-4b6c-bb6e-cf05c367cf60',
  pricing: {
    currency: 'USD',
  },
};

AirService.shop(params)
  .then(
    res => console.log(res),
    err => console.log(err)
  );
```

**Comment by dchertousov** [comment]
Hello!

The library was never tested in 1P, as we never had such tasks and access to 1P PCCs.

Feel free to fork, build and try it by yourself.

You may need to run the test script with Babel-node (see babel-cli).

Comments about 1P compatibility will be most appreciated!

**Comment by joshdhenry** [comment]
I will fork it and see how it goes.  Thanks for the response and for this great tool!

**Comment by dchertousov** [comment]
@joshdhenry have you tried it with 1P? Did you have any issues?
Do you think, you may extend configuration to include provider?

**Comment by joshdhenry** [comment]
@dchertousov I was able to get some basic air requests working by changing all instances of the string "1G" to "1P". There may have been a bit more to it, but it wasn't that difficult to get it working.  I didn't end up using this in my project, though, so you may see issues I didn't notice.

**Comment by dchertousov** [comment]
@joshdhenry Thank you for the response! Could I ask you what library did you finally used and why have you chosen them over uapi-json?

**Comment by dchertousov** [comment]
@InkFaust please investigate the issue and prepare your questions.
I think we should make provider configurable with default value of `1G`

**Comment by InkFaust** [comment]
@dchertousov got it.
I think configurable param for provider selection should be declared within `auth` object as optional param with `1G` default value.

**Comment by dchertousov** [comment]
Closed by #355

**Comment by dchertousov** [comment]
Hello @joshdhenry 

We have published today a new version 1.0.3 of uapi-json, allowing provider configuration.
Thank for your report
